test/irb/test_input_method.rb

```diff
@@ -8,10 +8,22 @@ class TestRelineInputMethod < Test::Unit::TestCase
     def setup
       @conf_backup = IRB.conf.dup
       IRB.conf[:LC_MESSAGES] = IRB::Locale.new
+
+      # RelineInputMethod#initialize calls IRB.set_encoding, which mutates standard input/output's encoding
+      # so we need to make sure we set them back
+      @original_encodings = {
+        STDIN => [STDIN.external_encoding, STDIN.internal_encoding],
+        STDOUT => [STDOUT.external_encoding, STDOUT.internal_encoding],
+        STDERR => [STDERR.external_encoding, STDERR.internal_encoding],
+      }
     end
 
     def teardown
       IRB.conf.replace(@conf_backup)
+
+      @original_encodings.each do |io, (external_encoding, internal_encoding)|
+        io.set_encoding(external_encoding, internal_encoding)
+      end
```

Comment:
I don't think testing the default values for the encodings here is a good idea.

Reply:
Maybe not specifically in this test, but as a global `teardown` callback for `irb` tests. Otherwise it's easy to break the Ruby CI again if someday a new test case accidentally changed encoding again without knowing it.